listServer.js

```diff
@@ -72,16 +76,27 @@ function denyRequest (req, res) {
 // --- Functions --- //
 // API: Returns JSON list of servers.
 function apiGetServerList(req, res) {
-	if(req.method != "GET") {
+	if(req.method != "POST") {
 		console.log(`[WARN] Denied request from ${req.ip}; expected GET but got ${req.method} instead!`)		
 		return res.sendStatus(400)
 	}
+	
+	if(req.body.serverKey === undefined || req.body.serverKey != secureLSKey)
+	{
+		console.log(`[WARN] Keys do not match ${req.body.serverKey} ${secureLSKey}`);
+		return res.sendStatus(400)
+	}
+	else
+	{
+		//Shows if keys match for those getting list server details.
+		console.log(`[WARN] Keys match ${req.body.serverKey} ${secureLSKey}`);
+	}
 
 	// A client wants the server list. Compile it and send out via JSON.
 	var serverList = [];
 	
 	for (var i = 0, len = knownServers.length; i < len; i++) {
-		serverList.push({ 'ip': knownServers[i].ip, 'name': knownServers[i].name, 'port': parseInt(knownServers[i].port), 'players': parseInt(knownServers[i].players) })
+		serverList.push({ 'ip': knownServers[i].ip, 'name': knownServers[i].name, 'port': parseInt(knownServers[i].port), 'players': parseInt(knownServers[i].players), 'serverKey': knownServers[i].serverKey })
```

Comment:
Why are we sending the serverKey to the client? Is this going to pose as a security risk?